### PR TITLE
Fix/CC-2558: Use correct back button urls on failed validation

### PR DESCRIPTION
**Pull request by mwejuli-ch**
### JIRA link

https://companieshouse.atlassian.net/browse/ROE-2558

### Change description

Resolve issues found in QA testing

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: test/controllers/beneficial.owner.type.controller.spec.ts [hunk 1]
```diff
@@ -412,8 +412,8 @@ describe("BENEFICIAL OWNER TYPE controller", () => {
     });
 
     const expectTextToContainBOTypePageURL = (text: string) => {
-      expect(mockGetUrlWithParamsToPath).toHaveBeenCalledTimes(1);
-      expect(mockGetUrlWithParamsToPath.mock.calls[0][0]).toEqual(config.ACTIVE_SUBMISSION_BASE_PATH);
+      expect(mockGetUrlWithParamsToPath).toHaveBeenCalledTimes(2);
+      // expect(mockGetUrlWithParamsToPath.mock.calls[0][0]).toEqual(config.ACTIVE_SUBMISSION_BASE_PATH);
```
Comment (dlloyd1-ch) [inline test/controllers/beneficial.owner.type.controller.spec.ts:416]: Line needs to either be removed or uncommented.

File: test/controllers/beneficial.owner.type.controller.spec.ts [hunk 2]
```diff
@@ -564,8 +564,8 @@ describe("BENEFICIAL OWNER TYPE controller", () => {
       expect(resp.status).toEqual(200);
       expect(resp.text).toContain(BENEFICIAL_OWNER_TYPE_PAGE_HEADING_SOME_IDENTIFIED);
       expect(resp.text).toContain(ErrorMessages.MUST_ADD_MANAGING_OFFICER);
-      expect(mockGetUrlWithParamsToPath).toHaveBeenCalledTimes(1);
-      expect(mockGetUrlWithParamsToPath.mock.calls[0][0]).toEqual(config.ACTIVE_SUBMISSION_BASE_PATH);
+      expect(mockGetUrlWithParamsToPath).toHaveBeenCalledTimes(2);
+      expect(mockGetUrlWithParamsToPath.mock.calls[0][0]).toEqual(config.BENEFICIAL_OWNER_STATEMENTS_WITH_PARAMS_URL);
```
Comment (dlloyd1-ch) [inline test/controllers/beneficial.owner.type.controller.spec.ts:568]: Not quites sure what's happening here. Can you clarify why the URL now refers to a different page? Same with the tests below.

File: test/controllers/beneficial.owner.type.controller.spec.ts (hunk 1 above)
Comment (mwestacott) [inline test/controllers/beneficial.owner.type.controller.spec.ts:416]: Should this line be commented out? I'd expect not.

File: test/controllers/managing.officer.corporate.controller.spec.ts [hunk 3]
```diff
@@ -134,17 +135,52 @@ describe("MANAGING_OFFICER CORPORATE controller", () => {
       expect(resp.text).toContain(SHOW_INFORMATION_ON_PUBLIC_REGISTER);
     });
 
-    test(`renders the ${MANAGING_OFFICER_CORPORATE_PAGE} page without public register jurisdiction field`, async () => {
+    /**
+     * This test should ideally be working but will skip it for now until a fix is worked out
+     */
```
Comment (mwestacott) [inline test/controllers/managing.officer.corporate.controller.spec.ts:140]: We should fix this now and not leave this to later.

File: test/controllers/overseas.entity.due.diligence.controller.spec.ts [hunk 4]
```diff
@@ -139,6 +139,52 @@ describe("OVERSEAS_ENTITY_DUE_DILIGENCE controller", () => {
       expect(resp.status).toEqual(500);
       expect(resp.text).toContain(SERVICE_UNAVAILABLE);
     });
+
+    test(`renders the ${OVERSEAS_ENTITY_DUE_DILIGENCE_PAGE} page with correct backlink url when feature flag is on`, async () => {
```
Comment (mwestacott) [inline test/controllers/overseas.entity.due.diligence.controller.spec.ts:143]: Would be helpful to name the feature flag in the test description (or as a comment on line 144).

File: test/controllers/overseas.entity.due.diligence.controller.spec.ts [hunk 5]
```diff
@@ -139,6 +139,52 @@ describe("OVERSEAS_ENTITY_DUE_DILIGENCE controller", () => {
...
+      expect(resp.text).toContain(ALL_THE_OTHER_INFORMATION_ON_PUBLIC_REGISTER);
+      expect(resp.text).toContain(NEXT_PAGE_URL);
+      expect(resp.text).toContain(BACK_BUTTON_CLASS);
+      expect(resp.text).toContain(OVERSEAS_ENTITY_DUE_DILIGENCE_PARTNER_NAME_HINT_TEXT);
+      expect(resp.text).toContain(SAVE_AND_CONTINUE_BUTTON_TEXT);
+      expect(resp.text).not.toContain(PAGE_TITLE_ERROR);
+      expect(mockIsActiveFeature).toHaveBeenCalledTimes(1);
+    });
+
+    test(`renders the ${OVERSEAS_ENTITY_DUE_DILIGENCE_PAGE} page with correct backlink url when feature flag is off`, async () => {
```
Comment (mwestacott) [inline test/controllers/overseas.entity.due.diligence.controller.spec.ts:166]: Same here. Which feature flag? I can guess this now for this PR but might not be so obvious when reading code back later on.

File: test/controllers/beneficial.owner.type.controller.spec.ts (hunk 1 above)
Comment (mwejuli-ch) [reply test/controllers/beneficial.owner.type.controller.spec.ts:416]: This is now captured using the `config.BENEFICIAL_OWNER_STATEMENTS_WITH_PARAMS_URL` which includes the `ACTIVE_SUBMISSION_BASE_PATH` so will remove the commented out line

File: test/controllers/beneficial.owner.type.controller.spec.ts (hunk 2 above)
Comment (mwejuli-ch) [reply test/controllers/beneficial.owner.type.controller.spec.ts:568]: As mentioned in the comment above, this is now captured in the `_WITH_PARAMS_URL` assertion

File: test/controllers/managing.officer.corporate.controller.spec.ts (hunk 3 above)
Comment (mwejuli-ch) [reply test/controllers/managing.officer.corporate.controller.spec.ts:140]: Sure..

File: test/controllers/overseas.entity.due.diligence.controller.spec.ts (hunk 5 above)
Comment (mwejuli-ch) [reply test/controllers/overseas.entity.due.diligence.controller.spec.ts:166]: have now specified the REDIS_removal  feature flag in the test description

File: test/controllers/overseas.entity.due.diligence.controller.spec.ts (hunk 4 above)
Comment (mwejuli-ch) [reply test/controllers/overseas.entity.due.diligence.controller.spec.ts:143]: have now specified the REDIS_removal feature flag in the test description

File: test/controllers/managing.officer.corporate.controller.spec.ts (hunk 3 above)
Comment (mwejuli-ch) [reply test/controllers/managing.officer.corporate.controller.spec.ts]: > We should fix this now and not leave this to later.

Fixed now.

Comment (mwestacott) [reply test/controllers/managing.officer.corporate.controller.spec.ts]: Nice. Guessing this was always broken. Must have worked by accident when running the app or the issue just not spotted.

**Review by mwestacott — APPROVED** [review]

**Review by dlloyd1-ch — APPROVED** [review]

**Pull request merged**